go/types/codec.go

```diff
@@ -16,63 +16,41 @@ const initialBufferSize = 2048
 
 func EncodeValue(v Value) chunks.Chunk {
 	w := newBinaryNomsWriter()
-	enc := newValueEncoder(w)
-	enc.writeValue(v)
-
-	c := chunks.NewChunk(w.data())
-	if cacher, ok := v.(hashCacher); ok {
-		assignHash(cacher, c.Hash())
-	}
-
-	return c
+	v.writeTo(w)
```

Comment:
> do we need to writeTo now? can we just hand over the internally buffer for v rather than allocating again?

We still need it since Type is not yet backed by a []byte

> to wit: does EncodeValue even make sense now?

Same here.

Once all values are backed by []byte we can revisit. Adding todo

Reply:
Also, the primitives...